### PR TITLE
Fix GitHub Actions based CI checks which have been failing

### PR DESCRIPTION
**Pull request by lhotari**
### Motivation

The current CI checks always fail. This PR attemps to fix this

### Modifications

- upgrade kind, helm chart and chart releaser versions
- disable podMonitor for remaining check
- make k8s 1.21 the baseline version
- upgrade application version to 2.10.4 since Zookeeper TLS tests didn't pass with 2.10.2

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -24,7 +24,7 @@ on:
       - '*'
 jobs:
   lint-test:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
```
[the remark this answers is not in the document]
Comment (lhotari) [reply .github/workflows/lint.yml:27]: Trying to find out the reason why things fail. btw. ubuntu-20.04 EoL is Apr 2025 so it's still valid.

[the remark this answers is not in the document]
Comment (lhotari) [reply .github/workflows/lint.yml]: Using 22.04 now

**Review by dave2wave — APPROVED** [review]
LGTM

**Comment by lhotari** [comment]
I think that the original reason for it to break is that ubuntu-latest switched to ubuntu-22.04 at some point. ubuntu-22.04 defaults to cgroup v2. The oldest k8s version to support cgroup v2 is k8s 1.19 . I'm not sure which kind version starts supporting cgroup v2. I also found out that for kind, you need to use supported versions of the kindest/node docker images which are documented in the kind release notes at https://github.com/kubernetes-sigs/kind/releases .

The reason to switch to 1.21 as the base line is the fact that the oldest k8s version supported by kind 0.20.0 is k8s 1.21. Anyways old k8s versions are end-of-life in many ways so there shouldn't be a good reason to keep support for very old k8s versions around.

**Pull request merged**